tfe/data_source_organizations.go

```diff
@@ -53,3 +56,63 @@ func dataSourceTFEOrganizationList(d *schema.ResourceData, meta interface{}) err
 
 	return nil
 }
+
+func adminOrgsPopulateFields(client *tfe.Client, d *schema.ResourceData) ([]string, map[string]string, error) {
+	names := []string{}
+	ids := map[string]string{}
+	log.Printf("[DEBUG] Listing all organizations (admin)")
+	options := tfe.AdminOrganizationListOptions{}
+	for {
+		orgList, err := client.Admin.Organizations.List(ctx, options)
```

Comment:
This is problematic at any larger scale, sadly. I hooked this up to an installation with several thousand organizations and it took so long I canceled it after 15 minutes.

Short of a specialized endpoint for making this really performant, I think the most practical thing we can do here is to use sparse fieldsets to minimize serialization and max out the page size, which is currently twenty 😱. That is, you'll need to make this URL what is used here (and after that, might as well add it to the other endpoint, too):

```
/api/v2/admin/organizations?fields[organization]=name&page[size]=100
```

This also tells us that query filtering is undoubtedly going to be the next step here to limit results (similar to `d/tfe_workspace.names`, but smarter), but we can save that for another day.